Library/AppEnvironment.swift

```diff
@@ -404,7 +448,14 @@ public struct AppEnvironment: AppEnvironmentType {
     var data: [String: Any] = [:]
 
     // swiftformat:disable wrap
-    data["apiService.oauthToken.token"] = env.apiService.oauthToken?.token
+
+    if let oauthToken = env.apiService.oauthToken?.token {
+      // Try to save to the keychain, but if that fails, save to user defaults
+      if !self.storeOAuthTokenToKeychain(oauthToken, forUserId: env.currentUser?.id) {
```

Comment:
Again, this will fail if the feature flag is off, and it will resume saving it to user defaults.